js/adapt-stateful-session.js

```diff
@@ -79,6 +79,24 @@ define([
 			this.listenTo(Adapt.blocks, 'change:_isComplete', this.onBlockComplete);
 			this.listenTo(Adapt.course, 'change:_isComplete', this.onCompletion);
 			this.listenTo(Adapt, 'assessment:complete', this.onAssessmentComplete);
+			this.listenTo(Adapt, 'app:resetSession', this.onResetSession);
+		},
+
+		removeEventListeners: function () {
+			$(window).off('unload', this._onWindowUnload);
+
+			if (this._shouldStoreResponses) {
+				this.stopListening(Adapt.components, 'change:_isInteractionComplete', this.onQuestionComponentComplete);
+			}
+
+			if(this._shouldRecordInteractions) {
+				this.stopListening(Adapt, 'questionView:recordInteraction', this.onQuestionRecordInteraction);
+			}
+
+			this.stopListening(Adapt.blocks, 'change:_isComplete', this.onBlockComplete);
+			this.stopListening(Adapt.course, 'change:_isComplete', this.onCompletion);
+			this.stopListening(Adapt, 'assessment:complete', this.onAssessmentComplete);
+			this.stopListening(Adapt, 'app:resetSession', this.onResetSession);
```

Comment:
Could lines 87–99 be changed to simply:
```javascript
this.stopListening();
````
It looks like it's unnecessary to explicitly name all the callbacks here.